spec/game_spec.rb

```diff
@@ -1,88 +1,141 @@
+class FakeMessages
+  def current(player:)
+    "current #{player}"
   end
 
-  def display(formatted_string)
-    self.log = formatted_string
+  def winning(player:)
+    "winning #{player}"
   end
+end
 
-  def get_input
+class GetOneInput
+  def get
     '1'
   end
 end
 
+class FakePresenter
+  attr_accessor :log
+
+  def initialize
+    @log = []
+  end
+
+  def present(board:, message:)
+    log << board.first
+    log << message
+  end
+end
+
 describe 'game' do
-  let(:user_interface) do
-    MockUserInterface.new
+  let(:symbols) { %w[O X].cycle }
+
+  let(:messages) { FakeMessages.new }
+
+  let(:input) { GetOneInput.new }
+
+  let(:presenter) { FakePresenter.new }
+
+  it 'Puts the current players move on the board if it is not full' do
+    board = MovesUntilFullBoard.new(1)
+    game = Game.new(
+      board: board,
+      symbols: symbols,
+      messages: messages,
+      input: input,
+      presenter: presenter
+    )
+    game.run
+    expect(board.latest_move).to eq %w[X 1]
   end
 
-  it 'Displays the game board' do
+  it 'Puts the next players move on the board if it is still not full' do
+    board = MovesUntilFullBoard.new(2)
     game = Game.new(
-      board: MockBoard.new(0),
-      symbols: %w[O X].cycle,
-      user_interface: user_interface
+      board: board,
+      symbols: symbols,
+      messages: messages,
+      input: input,
+      presenter: presenter
     )
     game.run
-    expect(user_interface.log).to eq [['']]
+    expect(board.latest_move).to eq %w[O 1]
   end
 
-  it 'Puts the current players move on the board if it is not full' do
+  it 'Displays the first player message' do
+    board = MovesUntilFullBoard.new(2)
     game = Game.new(
-      board: MockBoard.new(1),
-      symbols: %w[O X].cycle,
-      user_interface: user_interface
+      board: board,
+      symbols: symbols,
+      messages: messages,
+      input: input,
+      presenter: presenter
     )
     game.run
-    expect(user_interface.log).to eq [['X 1']]
+    expect(presenter.log.include?('current X')).to eq true
   end
 
-  it 'Puts the next players move on the board if it is still not full' do
+  it 'Displays the second player message' do
+    board = MovesUntilFullBoard.new(2)
     game = Game.new(
-      board: MockBoard.new(2),
-      symbols: %w[O X].cycle,
-      user_interface: user_interface
+      board: board,
+      symbols: symbols,
+      messages: messages,
+      input: input,
+      presenter: presenter
     )
     game.run
-    expect(user_interface.log).to eq [['O 1']]
+    expect(presenter.log.include?('current O')).to eq true
+  end
+
+  it 'Displays the board' do
+    board = MovesUntilFullBoard.new(2)
+    game = Game.new(
+      board: board,
+      symbols: symbols,
+      messages: messages,
+      input: input,
+      presenter: presenter
+    )
+    game.run
+    expect(presenter.log.include?('board')).to eq true
   end
 
   it 'Displays the winning player' do
+    board = MovesUntilFullBoard.new(3)
```

Comment:
using let could clean up this and reduce duplication of test instantiation

Reply:
It just got a lot cleaner, thank you!